pennylane/devices/default_qubit.py

```diff
@@ -497,6 +516,7 @@ def expval(self, observable, shot_range=None, bin_size=None):
             ) and observable.name == "Hamiltonian"
 
             if backprop_mode:
+                # TODO[dwierichs]: This branch is not adapted to broadcasting yet
```

Comment:
I'm not sure if this is what you want, but you could check for a broadcasting dimension, and if present, set `backprop_mode = False`. Or, would you rather `broadcast_expand` happens in this regime?

Reply:
I'm not sure how to fix this best intermediately. In the long run, adapting this branch to broadcasting should be doable (hopefully :sweat_smile: )
Maybe we should just add a check here that no broadcasting is used? :)

Reply:
I think it is fine, because we do not have any logic that automatically uses parameter broadcasting yet! 
Even the next PR on the stack that introduces broadcasting to the parameter-shift rule will only do so via a kwarg, which defaults to `False`. I think the feature support gap here could be closed with a very minor PR, btw :) I just wouldn't want to grow this one even bigger :see_no_evil:

Reply:
perfect!